gecko/crawlers/hana.py

```diff
@@ -0,0 +1,282 @@
+                levels = tuple(int(level) for level in child['id'].replace('row_', '').split('_')[1:-1])
+                if is_directory(child):
+                    parents[levels] = url
+                if levels[:-1]:
+                    file_hierarchy[url] = parents[levels[:-1]]
+                else:
+                    file_hierarchy[url] = file_path
+
+    return file_hierarchy
+
+
+def crawl_reference_page(contents, file_path, sections, title, up):
+    lvl1 = [{
+        'title': title,
+        'url': file_path
+    }]
+    if title in ['Concepts', 'Data types', 'Functional', 'Core',
+                 'Experimental features', 'External adapters',
+                 'Configuration options',
+                 'Assertions', 'Details']:
+        lvl1 += [{'title': 'Reference documentation', 'url': urljoin(file_path, 'modules.html')}]
+    copyright_notice = contents.select_one('.copyright')
+    if copyright_notice:
+        copyright_notice.decompose()
+    content = []
+    anchors = []
+    anchor_found = False
+    for child in contents.children:
+        if isinstance(child, Tag):
+            if child.name == 'a' and child.has_attr('id'):
+                child_id = child.get('id')
+                anchors.append({'id': child_id})
+                anchor_found = True
+            elif child.name == 'table' and child.select_one('.heading'):
+                heading, *body = child.select('tr')
+                link = heading.select_one('a').get('name')
+                section_title = sanitize_title(heading.get_text("\n"))
+                section_body = '\n'.join([x.get_text("\n") for x in body])
+                if section_body.find('More...') != -1:
+                    continue
+                url = file_path + '#' + link
+                sections[url] = {
+                    'content': section_body,
+                    'lvls': [{
+                        'title': section_title,
+                        'url': url
+                    }] + lvl1,
+                    'up': up
+                }
+            elif anchor_found:
+                if child.name in ['h2', 'h3', 'h4', 'h5', 'h6']:
+                    anchors[-1]['title'] = sanitize_title(child.find(text=True, recursive=False))
+                    continue
+                memproto = child.select_one('.memproto')
+                if memproto:
+                    if 'title' not in anchors[-1]:
+                        anchors[-1]['title'] = sanitize_title(memproto.get_text("\n"))
+                    doc = child.select_one('.memdoc')
+                    anchors[-1]['doc'] = doc.get_text("\n")
+                else:
+                    anchors[-1]['doc'] = child.get_text("\n")
+                anchor_found = False
+            elif child.name not in ['h2', 'h3', 'h4', 'h5', 'h6']:
+                content.append(child.get_text("\n"))
+    # merge all content
+    content = '\n'.join(content)
+    if content.find('More...') == -1:
```

Comment:
Will this only affect header pages?

Reply:
It affects header pages and those two:
- https://www.boost.org/doc/libs/1_82_0/libs/hana/doc/html/namespaceboost_1_1hana_1_1literals.html
- https://www.boost.org/doc/libs/1_82_0/libs/hana/doc/html/namespaceboost_1_1hana.html

But that's what we want!